lib/node_modules/@stdlib/lapack/base/dlaset/lib/base.js

```diff
@@ -0,0 +1,128 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var min = require( '@stdlib/math/base/special/min' );
+
+
+// MAIN //
+
+/**
+* Initializes an `M` by `N` matrix `A` to `beta` on the diagonal and `alpha` on the off-diagonals.
+*
+* @private
+* @param {string} order - storage layout
+* @param {string} uplo - specifies whether the upper or lower triangular part of matrix `A` is supplied
+* @param {NonNegativeInteger} M - number of rows in matrix `A`
+* @param {NonNegativeInteger} N - number of columns in matrix `A`
+* @param {number} alpha - scalar constant to which off-diagonal elements are set
+* @param {number} beta - scalar constant to which diagonal elements are set
+* @param {Float64Array} A - input matrix
+* @param {PositiveInteger} LDA - stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`)
+* @param {NonNegativeInteger} offsetA - starting index for `A`
+* @returns {Float64Array} `A`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ] );
+*
+* dlaset( 'row-major', 'lower', 3, 3, 3.1, 3.7, A, 3, 0 );
+* // A => <Float64Array>[ 3.7, 0.0, 0.0, 3.1, 3.7, 0.0, 3.1, 3.1, 3.7 ]
+*/
+function dlaset( order, uplo, M, N, alpha, beta, A, LDA, offsetA ) {
+	var i;
+	var j;
+
+	if ( uplo === 'upper' ) {
+		if ( order === 'column-major' ) {
+			for ( j = 1; j < N; j++ ) {
+				for ( i = 0; i < min( j - 1, M ); i++ ) {
+					A[ offsetA + ( j * LDA ) + i ] = alpha;
+				}
+			}
+			for ( i = 0; i < min( M, N ); i++ ) {
+				A[ offsetA + ( i * LDA ) + i ] = beta;
+			}
+			return A;
+		}
+		// order === 'row-major'
+		for ( j = 1; j < N; j++ ) {
+			for ( i = 0; i < min( j - 1, M ); i++ ) {
+				A[ offsetA + ( i * LDA ) + j ] = alpha;
+			}
+		}
+		for ( i = 0; i < min( M, N ); i++ ) {
+			A[ offsetA + ( i * LDA ) + i ] = beta;
+		}
+		return A;
+	}
+	if ( uplo === 'lower' ) {
+		if ( order === 'column-major' ) {
+			for ( j = 0; j < min( M, N ); j++ ) {
+				for ( i = j + 1; i < M; i++ ) {
+					A[ offsetA + ( j * LDA ) + i ] = alpha;
+				}
+			}
+			for ( i = 0; i < min( M, N ); i++ ) {
+				A[ offsetA + ( i * LDA ) + i ] = beta;
+			}
+			return A;
+		}
+		// order === 'row-major'
+		for ( j = 0; j < min( M, N ); j++ ) {
+			for ( i = j + 1; i < M; i++ ) {
+				A[ offsetA + ( i * LDA ) + j ] = alpha;
+			}
+		}
+		for ( i = 0; i < min( M, N ); i++ ) {
+			A[ offsetA + ( i * LDA ) + i ] = beta;
+		}
+		return A;
+	}
+	// All of the matrix `A`
+	if ( order === 'column-major' ) {
```

Comment:
These loops should definitely support loop interchange, thus allowing consolidation into a single set of nested loops.

Reply:
@Pranavchiku You can have a look at the most recent changes to the `dlacpy` PR. https://github.com/stdlib-js/stdlib/pull/2548

Reply:
As in that PR, I suggest splitting this implementation up into separate sub-functions.